src/ngraph/op/strided_slice.cpp

```diff
@@ -139,19 +175,40 @@ void op::v1::StridedSlice::validate_and_infer_types()
 
     auto begin_const = as_type_ptr<op::Constant>(input_value(1).get_node_shared_ptr());
     auto end_const = as_type_ptr<op::Constant>(input_value(2).get_node_shared_ptr());
-    auto strides = as_type_ptr<op::Constant>(input_value(3).get_node_shared_ptr());
+    auto strides_const = as_type_ptr<op::Constant>(input_value(3).get_node_shared_ptr());
 
-    if (begin_const && end_const && strides)
+    if (begin_const && end_const && strides_const && data_rank.is_static())
     {
+        auto begin = begin_const->cast_vector<int64_t>();
+        auto end = end_const->cast_vector<int64_t>();
+        auto strides = strides_const->cast_vector<int64_t>();
+        const auto begin_mask = convert_mask_to_axis_set(get_begin_mask());
+        const auto end_mask = convert_mask_to_axis_set(get_end_mask());
+
+        // Handle a case when begin/end/stride size is less than data rank
+        // and begin/end mask are passed
```

Comment:
Why does the StridedSlice have inputs with begin/end masks those rank is less than input data rank? Our specification does not explicitly states that but I believe that such a case should not be accepted. In this case it will be not clear how to apply striding: from the left or from the right? Could you clarify from where such a case comes from?

Reply:
You are right that spec does not specify such cases in clear way.
I've moved all adjustment between `Slice-10` and `StridedSlice:v1` interfaces to `Slice-10` implementation.
As a result I marked all your requests connected with `extend_vector_by_value` for `StridedSlice` as resolved (out of date).